lochness/mindlamp/__init__.py

```diff
@@ -150,23 +156,16 @@ def sync(Lochness: 'lochness.config',
 
             # confirm it's a new file
             if Path(dst).is_file():
-                with open(dst, 'r') as existing_file:
-                    # [-1] at the end -> last timepoint data
-                    existing_data = json.load(existing_file)[-1]
-
-                new_data = function_to_execute(
-                    LAMP, subject_id,
-                    from_ts=existing_data['timestamp'],
-                    to_ts=existing_data['timestamp'])[0]
-
-                existing_data_json = json.dumps(existing_data, sort_keys=True)
-                new_data_json = json.dumps(new_data, sort_keys=True)
-
-                if existing_data_json == new_data_json:
+                checksum_file = dst.parent / f'.check_sum_{dst.name}'
+                if checksum_file.is_file():
                     logger.debug(f'{data_name} data exist for {date_str} '
                                  '- not downloading')
                     continue
 
+                prev_file_sha256 = get_sha(dst)
+            else:
+                prev_file_sha256 = ''
+
```

Comment:
In the above, I see that `dst` is just a file name, not the file content. If that is right, between previous and new `dst`, only the `date_str` are different. If that is right, why not use just `date_str` as the basis for identifying new data?

Reply:
`get_sha` function in `lochness.utils.checksum` actually reads the file.